lib/vrf.c

```diff
@@ -997,17 +997,29 @@ const char *vrf_get_default_name(void)
 int vrf_bind(vrf_id_t vrf_id, int fd, const char *name)
 {
 	int ret = 0;
+	struct vrf *vrf;
 	struct interface *ifp;
 
-	if (fd < 0 || name == NULL)
+	if (fd < 0 || name == NULL || vrf_is_backend_netns()
+	    || (strcmp(name, "default") == 0))
```

Comment:
at startup, vrf_is_backend information may not be available for daemon.
actually, imagine vrf_bind is called by BGP whereas it is not yet connected with zebra.

I think more something that checks for vrf_id parameter.
if vrf_id == VRF_UNKNOWN : vrf is not enabled
if vrf_id == real value : vrf is enabled ( VRF_DEFAULT is ok too).

Reply:
OK in this test I am trying to return early for default vrfs or any vrf if we are using netns-based vrfs so I can see I want 
if (vrf_id == VRF_UNKNOWN || vrf_id == vrf_default)
    return;
but does this catch the netns case?

Reply:
added check for vrf_id == VRF_UNKNOWN